app/controllers/AssessmentScheduleController.scala

```diff
@@ -0,0 +1,427 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import java.io.UnsupportedEncodingException
+import java.net.URLDecoder
+
+import com.github.nscala_time.time.OrderingImplicits.LocalDateOrdering
+import common.FutureEx
+import connectors.AssessmentScheduleExchangeObjects._
+import connectors.ExchangeObjects.AllocationDetails
+import connectors.{ CSREmailClient, EmailClient }
+import model.AssessmentScheduleCommands.ApplicationForAssessmentAllocationResult
+import model.AssessmentScheduleCommands.Implicits.ApplicationForAssessmentAllocationResultFormats
+import model.Commands.Implicits.applicationAssessmentFormat
+import model.Exceptions.NotFoundException
+import model.{ Commands, ProgressStatuses }
+import org.joda.time.LocalDate
+import play.api.libs.json.{ JsValue, Json }
+import play.api.mvc.{ Action, AnyContent, Result }
+import repositories.AssessmentCentreLocation.assessmentCentreVenueFormat
+import repositories._
+import repositories.application._
+import services.AuditService
+import uk.gov.hmrc.play.microservice.controller.BaseController
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.Try
+
+object AssessmentScheduleController extends AssessmentScheduleController {
+  // val aaRepository: AssessmentCentreAllocationMongoRepository = assessmentCentreAllocationRepository
+  val acRepository = AssessmentCentreYamlRepository
+  val aRepository: GeneralApplicationMongoRepository = applicationRepository
+  // val otRepository: OnlineTestMongoRepository = onlineTestRepository
+  val auditService = AuditService
+  // val pdRepository: PersonalDetailsRepository = personalDetailsRepository
+  // val cdRepository: ContactDetailsRepository = contactDetailsRepository
+  val emailClient = CSREmailClient
+  // val aaService = AssessmentCentreService
+  // val assessmentScoresService = AssessorAssessmentScoresService
+}
+
+trait AssessmentScheduleController extends BaseController {
+  // val aaRepository: AssessmentCentreAllocationRepository
+  val acRepository: AssessmentCentreRepository
+  val aRepository: GeneralApplicationRepository
+  // val otRepository: OnlineTestRepository
+  val auditService: AuditService
+  // val pdRepository: PersonalDetailsRepository
+  // val cdRepository: ContactDetailsRepository
+  val emailClient: EmailClient
+  // val aaService: AssessmentCentreService
+  // val assessmentScoresService: AssessmentCentreScoresService
+
+  def getAssessmentScheduleDatesByRegion(region: String): Action[AnyContent] = Action.async { implicit request =>
+     acRepository.assessmentCentreCapacities.map { schedule =>
+       val toRet = schedule
+         .filter(location => location.locationName == region)
```

Comment:
This is what I say in the previous PR, we use "location" although "region" may be better. We should be consistent. I am fine if we want to use region.